plugins/inputs/procstat/process_linux.go

```diff
@@ -0,0 +1,25 @@
+package procstat
+
+import (
+	"time"
+
+	"github.com/shirou/gopsutil/cpu"
+	"github.com/shirou/gopsutil/process"
+)
+
+type Process interface {
+	PID() PID
+	Tags() map[string]string
+
+	IOCounters() (*process.IOCountersStat, error)
+	MemoryInfo() (*process.MemoryInfoStat, error)
+	MemoryMaps(bool) (*[]process.MemoryMapsStat, error)
```

Comment:
Instead of one interface per platform, there should only be one interface and then platform specific implementations of the function.

You can add the new function to `type Proc struct`, but instead of implementing for all known platforms, do one implementation for Linux and a do nothing implementation for not linux.

Finally, because the MemoryMapsStat struct varies so much in gopsutil, I would probably make the function more specific.  On non-linux platforms you could define and return a error like `ErrNotImplementedError`:
```
MemoryPSS() (uint64, error)
```

An alternative to above would be wrapping the []process.MemoryMapsStat in an separate type for linux/non-linux with functions to retrieve the various fields.